mala/targets/calculation_helpers.py

```diff
@@ -62,8 +62,25 @@ def fermi_function(energy, fermi_energy, temperature):
         Value of the Fermi function.
 
     """
-    return 1.0 / (1.0 + np.exp((energy - fermi_energy) /
-                               (kB * temperature)))
+    exponent = (energy - fermi_energy) / (kB * temperature)
+
+    # Maximum exponent that will not result in inf when performing
+    # np.exp() (for double values only).
+    # Very large exponents result in very large values, and if we don't check
+    # for overflow, infinity. Cutting this back to the maximum float value
+    # (~ 1e308) is more then enough, since we divide by the exponent, and
+    # therefore get zero (or machine precision within zero) either way.
+    # Adding this check removes overflow warnings.
+    # The overhead for inference should be minimal.
+    # Since this function works both for arrays and scalars,
+    # we have to check which maximum to use.
+    if type(energy) == np.ndarray:
+        max_exponent = np.log(np.finfo(exponent.dtype.name).max)
+        exponent[np.argwhere(exponent > max_exponent)] = max_exponent
+    else:
+        exponent = min(exponent, np.log(sys.float_info.max))
```

Comment:
Here you assume that `exponent` is `float64` (since this is what Python's built-in float defaults to and what `sys.float_info` talks about).

```py
>>> type(1.0)
float
```

Note that `np.finfo` also accepts numbers, so all these are the same

```py
>>> np.finfo(1.0)
>>> np.finfo(type(1.0))
>>> np.finfo("float")
>>> np.finfo("float64")
```

As such, you don't need `sys.float_info` and can instead query `exponent` using it's actual type: `finfo(type(exponent))`.

Reply:
Oh I didn't know that, implemented it in 3f7bf58fa9f2a7544df45eecb158ce63f954dee1.